### PR TITLE
Recognize UnionPay as a separate brand.

### PR DESCRIPTION
**Pull request by dlubarov**
It seems more correct to consider UnionPay its own brand. Discover is just the proxy network.

### CODE REVIEW

**Comment by tamird** [comment]
:+1:

File: lib/plastic/core.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Plastic
-  BRANDS = [:visa, :mastercard, :american_express, :discover, :discover_diners, :jcb]
+  BRANDS = [:visa, :mastercard, :american_express, :discover, :discover_diners, :jcb, :unionpay]
```
Comment (zachmargolis) [inline lib/plastic/core.rb:2]: Our Java enum is `UNION_PAY`, correct? `unionpay` is not consistent with that, so we if ever try to upcase/downcase, we'll get different results

Comment (dlubarov) [inline lib/plastic/core.rb:2]: True but we have `mastercard`/`MASTER_CARD` already :confused:

File: spec/plastic/validations_spec.rb [hunk 2]
```diff
@@ -235,7 +235,7 @@ def format_expiration_year_as_track(year)
 
   describe "#valid_brand?" do
     it "is not valid if the brand is blank" do
-      plastic = Plastic.new(:pan => "0480020605154711", :expiration => "1312")
+      plastic = Plastic.new(:pan => "0480020605154711", :expiration => "1812")
```
Comment (rudle) [inline spec/plastic/validations_spec.rb]: nit: if you express the expiry as `1.year.from.now.strftime("%y%m")` you won't regret this change in 4 years.

Comment (tamird) [inline spec/plastic/validations_spec.rb]: no activesupport in da house

Comment (dlubarov) [inline spec/plastic/validations_spec.rb]: `(Date.today >> 12).strftime("%y%m")` sound okay?

Comment (tamird) [inline spec/plastic/validations_spec.rb]: sgtm

Comment (dlubarov) [inline spec/plastic/validations_spec.rb]: Done

**Pull request merged**